azurerm/provider.go

```diff
@@ -143,6 +143,7 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_virtual_machine":                       dataSourceArmVirtualMachine(),
 			"azurerm_virtual_network":                       dataSourceArmVirtualNetwork(),
 			"azurerm_virtual_network_gateway":               dataSourceArmVirtualNetworkGateway(),
+			"azurerm_application_insights":                  dataSourceArmApplicationInsights(),
```

Comment:
_minor_ could we sort this alphabetically to match the rest of this list? it helps when skimming it

Reply:
Ah, good point. I did an alphabetical sort of this list and there actually many elements out of order:

```diff
diff --git a/azurerm/provider.go b/azurerm/provider.go
index 13b180ea..842c4630 100644
--- a/azurerm/provider.go
+++ b/azurerm/provider.go
@@ -89,27 +89,28 @@ func Provider() terraform.ResourceProvider {
 		},
 
 		DataSourcesMap: map[string]*schema.Resource{
-			"azurerm_azuread_application":                   dataSourceArmAzureADApplication(),
-			"azurerm_azuread_service_principal":             dataSourceArmActiveDirectoryServicePrincipal(),
 			"azurerm_api_management":                        dataSourceApiManagementService(),
-			"azurerm_application_security_group":            dataSourceArmApplicationSecurityGroup(),
-			"azurerm_app_service":                           dataSourceArmAppService(),
 			"azurerm_app_service_plan":                      dataSourceAppServicePlan(),
+			"azurerm_app_service":                           dataSourceArmAppService(),
+			"azurerm_application_insights":                  dataSourceArmApplicationInsights(),
+			"azurerm_application_security_group":            dataSourceArmApplicationSecurityGroup(),
+			"azurerm_azuread_application":                   dataSourceArmAzureADApplication(),
+			"azurerm_azuread_service_principal":             dataSourceArmActiveDirectoryServicePrincipal(),
 			"azurerm_batch_account":                         dataSourceArmBatchAccount(),
 			"azurerm_builtin_role_definition":               dataSourceArmBuiltInRoleDefinition(),
 			"azurerm_cdn_profile":                           dataSourceArmCdnProfile(),
 			"azurerm_client_config":                         dataSourceArmClientConfig(),
-			"azurerm_cosmosdb_account":                      dataSourceArmCosmosDBAccount(),
 			"azurerm_container_registry":                    dataSourceArmContainerRegistry(),
+			"azurerm_cosmosdb_account":                      dataSourceArmCosmosDBAccount(),
 			"azurerm_data_lake_store":                       dataSourceArmDataLakeStoreAccount(),
 			"azurerm_dev_test_lab":                          dataSourceArmDevTestLab(),
 			"azurerm_dns_zone":                              dataSourceArmDnsZone(),
 			"azurerm_eventhub_namespace":                    dataSourceEventHubNamespace(),
 			"azurerm_image":                                 dataSourceArmImage(),
-			"azurerm_key_vault":                             dataSourceArmKeyVault(),
-			"azurerm_key_vault_key":                         dataSourceArmKeyVaultKey(),
 			"azurerm_key_vault_access_policy":               dataSourceArmKeyVaultAccessPolicy(),
+			"azurerm_key_vault_key":                         dataSourceArmKeyVaultKey(),
 			"azurerm_key_vault_secret":                      dataSourceArmKeyVaultSecret(),
+			"azurerm_key_vault":                             dataSourceArmKeyVault(),
 			"azurerm_kubernetes_cluster":                    dataSourceArmKubernetesCluster(),
 			"azurerm_log_analytics_workspace":               dataSourceLogAnalyticsWorkspace(),
 			"azurerm_logic_app_workflow":                    dataSourceArmLogicAppWorkflow(),
@@ -120,8 +121,8 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_monitor_log_profile":                   dataSourceArmMonitorLogProfile(),
 			"azurerm_network_interface":                     dataSourceArmNetworkInterface(),
 			"azurerm_network_security_group":                dataSourceArmNetworkSecurityGroup(),
-			"azurerm_notification_hub":                      dataSourceNotificationHub(),
 			"azurerm_notification_hub_namespace":            dataSourceNotificationHubNamespace(),
+			"azurerm_notification_hub":                      dataSourceNotificationHub(),
 			"azurerm_platform_image":                        dataSourceArmPlatformImage(),
 			"azurerm_public_ip":                             dataSourceArmPublicIP(),
 			"azurerm_public_ips":                            dataSourceArmPublicIPs(),
@@ -143,7 +144,6 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_virtual_machine":                       dataSourceArmVirtualMachine(),
 			"azurerm_virtual_network":                       dataSourceArmVirtualNetwork(),
 			"azurerm_virtual_network_gateway":               dataSourceArmVirtualNetworkGateway(),
-			"azurerm_application_insights":                  dataSourceArmApplicationInsights(),
 		},
 
 		ResourcesMap: map[string]*schema.Resource{

```

Would you prefer leaving sorting all the lines for another PR and just place `azurerm_application_insights` where it should be, or should I send the change for all the lines? Sorting all the lines is a bit out of scope of this PR, IMO.

Reply:
Another PR for this would be great 🙂

Reply:
Ok, I'll mark this one as resolved and then send a subsequent PR once this gets merged with the sorting. Sounds good?

Reply:
Done in #2638

Reply:
> Ah, good point. I did an alphabetical sort of this list and there actually many elements out of order:

@inkel sorry I didn't realise they weren't in order! Thanks for #2638 in any case, I'll take a look now :)

Reply:
No problem! <3